views/physicalobject.py

```diff
@@ -24,9 +24,9 @@ def zoom(coordinates: Sequence[Vector2], scale_factor: float, zoom_level: float)
 
 
 def project(coordinates: Sequence[Vector3], normalVector: Vector3) -> list[Vector2]:
-    """Project the 3D coordinates onto a 2D plane."""
-
-    a, b, c = 2*math.sin(normalVector.x), 2*math.sin(normalVector.y), normalVector.z
+    """Project the 3D coordinates onto a 2D plane."""  
+    normalVector = normalVector.normalize()
+    a, b, c = normalVector.x, normalVector.y, normalVector.z
     return [
         Vector2(coordinate.x - a*(a*coordinate.x+ b*coordinate.y + c*coordinate.z) / math.sqrt(a**2 + b**2 + c**2), 
     coordinate.y - b*(a*coordinate.x+ b*coordinate.y + c*coordinate.z) / math.sqrt(a**2 + b**2 + c**2)) 
```

Comment:
Because the normalVector is normalized `math.sqrt(a**2 + b**2 + c**2)) ` always returns 1.